main/influxdb_watcher.go

```diff
@@ -136,6 +136,34 @@ func queryDB(broker *InfluxdbBroker, cmd string, ch chan []client.Result, cherr
 	ch <- response.Results
 }
 
+func (broker *InfluxdbBroker) checkDatabaseExists(timeout time.Duration) bool {
+	cmd := "SHOW DATABASES"
+	ch := make(chan []client.Result)
+	cherr := make(chan error)
+	go queryDB(broker, cmd, ch, cherr)
+	for {
+		select {
+		case response := <-ch:
+			for _, s := range response[0].Series {
+				for _, db := range s.Values {
+					if db[0] == "mon" {
+						log.Infof("Database %s exists", broker.MonascaDB)
+						return true
+					}
+				}
+				log.Infof("Database %s does not exist yet", broker.MonascaDB)
+				return false
+			}
+		case _ = <-cherr:
+			log.Infof("Database %s does not exist yet", broker.MonascaDB)
+			return false
+		case <-time.After(timeout):
+			log.Infof("Database %s does not exist yet", broker.MonascaDB)
```

Comment:
Since these two lines are exactly the same as 158 and 159, you can just delete 158 and 159 so the two cases are merged.

Reply:
Done